genesyscloud/user_group_role_common.go

```diff
@@ -73,88 +77,123 @@ func readSubjectRoles(subjectID string, authAPI *platformclientv2.AuthorizationA
 	for roleID, divs := range roleDivsMap {
 		role := make(map[string]interface{})
 		role["role_id"] = roleID
-		role["division_ids"] = divs
+		role["division_ids"] = addDivisionIdsSetToRole(d, divs, roleID, homeDivId)
 		roleSet.Add(role)
 	}
 	return roleSet, resp, nil
 }
 
-func updateSubjectRoles(ctx context.Context, d *schema.ResourceData, authAPI *platformclientv2.AuthorizationApi, subjectType string) diag.Diagnostics {
```

Comment:
The only change to this function was taking everything outside of the if statement `if d.HashChange("roles") {..}` and instead returning at the beginning if `!d.HasChange("roles")` or if `rolesConfig == nil`